packages/desktop-client/src/components/reports/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from 'react';
+
+import AnimatedLoading from '../../icons/AnimatedLoading';
+import { colors, styles } from '../../style';
+import { Block, View } from '../common';
+
+import type { ReportRouter } from './ReportRouter';
+
+export default function Reports() {
+  let [ReportRouterComponent, setReportRouter] = useState<
+    typeof ReportRouter | null
+  >(null);
+
+  useEffect(() => {
+    import(/* webpackChunkName: 'reports' */ './ReportRouter').then(module => {
+      setReportRouter(() => module.ReportRouter);
+    });
```

Comment:
IMO would be nicer to lazyload it via react-router: https://reactrouter.com/en/main/route/lazy

Reply:
ok will mark this PR as draft and port over to the new router APIs in a separate PR

Reply:
I took a look at the data router functionality, and it doesn’t seem like it would mix well with our current routing structure. In particular, we have a `StackedRoutes` component that allows us to render routes into modals stacked on top of one another. There doesn’t appear to be a way to map that usage of `<Routes location={someCustomLocation} />` onto the data router concept.